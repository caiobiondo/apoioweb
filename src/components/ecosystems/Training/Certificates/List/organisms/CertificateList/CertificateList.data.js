import gql from 'graphql-tag';

export const CertificateListQuery = gql`
  query CertificateListQuery(
    $sellerId: Int!
    $userName: String!
    $ciclo: Int
    $setor: Int
    $gerenciaMercado: Int
    $grupo: Int
    $papelDaConsultora: Int
    $canal: Int
    $gerenciaDeVendas: Int
    $regiao: Int
  ) {
    trainingCertificates(
      sellerId: $sellerId
      userName: $userName
      ciclo: $ciclo
      setor: $setor
      gerenciaMercado: $gerenciaMercado
      grupo: $grupo
      papelDaConsultora: $papelDaConsultora
      canal: $canal
      gerenciaDeVendas: $gerenciaDeVendas
      regiao: $regiao
    ) {
      id
      name
      thumbnail
      percentageOfCompletedCourse
      isCompleted
      downloadUrl
    }
  }
`;

export const CertificateListQueryOptions = {
  options(props) {
    return {
      variables: {
        sellerId: props.user.codigo,
        userName: props.user.nomeCompleto,
        ciclo:
          props.user.estrutura.ciclo &&
          props.user.estrutura.ciclo[0] &&
          props.user.estrutura.ciclo[0].numero,
        setor: props.user.estrutura.setor.codigo,
        gerenciaMercado: props.user.estrutura.gerenciaMercado.codigo,
        grupo: props.user.estrutura.codigo,
        papelDaConsultora: props.user.cdPapelAtivo,
        canal: props.user.cdCanalCaptacao,
        gerenciaDeVendas: props.user.estrutura.gerenciaVenda.codigo,
        regiao: props.user.estrutura.regiaoEstrategica.codigo,
      },
      forceFetch: true,
    };
  },

  props({ data }) {
    return {
      data,
      loading: data.loading,
      certificates: data.trainingCertificates,
    };
  },
};
