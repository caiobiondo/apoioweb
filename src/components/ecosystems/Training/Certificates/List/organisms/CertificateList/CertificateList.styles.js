import styled from 'styled-components';

export const List = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  padding: 30px 0;
`;

export const fullContainer = {
  height: '100%',
};
