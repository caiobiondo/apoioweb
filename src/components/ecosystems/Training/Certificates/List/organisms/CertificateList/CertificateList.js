import React, { Component } from 'react';
import Certificate from 'components/ecosystems/Training/Certificates/List/molecules/Certificate';
import PageMenu from 'components/ecosystems/Training/atoms/PageMenu/PageMenu';

import { List } from './CertificateList.styles';
import { Paper } from 'natura-ui';

const certificates = [
  {
    id: 51,
    name: 'Arrase nas vendas',
    tt: 'Arrase nas vendas',
    totalOfCourses: '6',
    totalOfCoursesCompleted: '1',
    thumbnail: 'http://treinamento-api-dev.natura.com.br:7000/category/imagethumbnail/51',
    percentageOfCompletedCourse: 16,
  },
  {
    id: 67,
    name: 'TESTE',
    thumbnail: 'http://treinamento-api-dev.natura.com.br:7000/category/imagethumbnail/67',
    totalOfCourses: '1',
    totalOfCoursesCompleted: '0',
    percentageOfCompletedCourse: 0,
  },
  {
    id: 68,
    name: 'Icone teste',
    thumbnail: 'http://treinamento-api-dev.natura.com.br:7000/category/imagethumbnail/68',
    totalOfCourses: '1',
    totalOfCoursesCompleted: '0',
    percentageOfCompletedCourse: 0,
  },
];

class CertificateList extends Component {
  render() {
    return (
      <Paper>
        <PageMenu />
        <List>
          {certificates.map((certificate, index) => (
            <Certificate key={index} index={index} certificate={certificate} />
          ))}
        </List>
      </Paper>
    );
  }
}

export default CertificateList;
