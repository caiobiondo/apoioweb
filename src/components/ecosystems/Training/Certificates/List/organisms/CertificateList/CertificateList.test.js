import React from 'react';
import { shallow } from 'enzyme';
import { CertificateList } from './CertificateList';

const setup = propOverrides => {
  const props = Object.assign(
    {
      user: {
        codigo: 1234,
        nomeCompleto: 'Teste User',
      },
      certificates: [],
    },
    propOverrides,
  );

  const result = shallow(<CertificateList {...props} />);

  return {
    props,
    result,
  };
};

describe('Training Certificate List', () => {
  it('renders correctly when loading', () => {
    // given
    // when
    const { result } = setup({ loading: true });

    // then
    expect(result).toMatchSnapshot();
  });

  it('renders correctly when the list is empty', () => {
    // given
    // when
    const { result } = setup({ loading: false });

    // then
    expect(result).toMatchSnapshot();
  });

  it('renders correctly when the list is not empty', () => {
    // given
    const mockCertificates = [
      {
        id: 1,
        name: 'Certiticate 1',
      },
      {
        id: 2,
        name: 'Certiticate 2',
      },
      {
        id: 3,
        name: 'Certiticate 3',
      },
    ];

    // when
    const { result } = setup({
      loading: false,
      certificates: mockCertificates,
    });

    // then
    expect(result).toMatchSnapshot();
  });
});
