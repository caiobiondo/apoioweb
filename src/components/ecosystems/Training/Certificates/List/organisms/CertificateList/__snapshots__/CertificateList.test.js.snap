// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Training Certificate List renders correctly when loading 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "height": "100%",
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <PageMenu />
  <EmptyList
    descriptionId="certificatesEmptyListDescription"
    icon="ico_list_add"
    titleId="myCertificatesEmptyList"
  />
</Paper>
`;

exports[`Training Certificate List renders correctly when the list is empty 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "height": "100%",
    }
  }
  transitionEnabled={true}
  zDepth={1}
>
  <PageMenu />
  <EmptyList
    descriptionId="certificatesEmptyListDescription"
    icon="ico_list_add"
    titleId="myCertificatesEmptyList"
  />
</Paper>
`;

exports[`Training Certificate List renders correctly when the list is not empty 1`] = `
<Paper
  circle={false}
  rounded={true}
  transitionEnabled={true}
  zDepth={1}
>
  <PageMenu />
  <CertificateListstyles__List>
    <withApollo(Certificate)
      certificate={
        Object {
          "id": 1,
          "name": "Certiticate 1",
        }
      }
      index={0}
      key="1"
      user={
        Object {
          "codigo": 1234,
          "nomeCompleto": "Teste User",
        }
      }
    />
    <withApollo(Certificate)
      certificate={
        Object {
          "id": 2,
          "name": "Certiticate 2",
        }
      }
      index={1}
      key="2"
      user={
        Object {
          "codigo": 1234,
          "nomeCompleto": "Teste User",
        }
      }
    />
    <withApollo(Certificate)
      certificate={
        Object {
          "id": 3,
          "name": "Certiticate 3",
        }
      }
      index={2}
      key="3"
      user={
        Object {
          "codigo": 1234,
          "nomeCompleto": "Teste User",
        }
      }
    />
  </CertificateListstyles__List>
</Paper>
`;
